sound/soc/sof/sof-pci-dev.c

```diff
@@ -152,6 +156,23 @@ int sof_pci_probe(struct pci_dev *pci, const struct pci_device_id *pci_id)
 	sof_pdata->dev = dev;
 	sof_pdata->fw_filename = desc->default_fw_filename;
 
+	sof_pdata->ipc_type = desc->ipc_default;
+
+	if (sof_pci_ipc_type != desc->ipc_default) {
```

Comment:
While this is true, it is not correct.
if the platform by default uses INTEL_CAVS (desc->ipc_default == SOF_INTEL_CAVS) and you don't specify any options to change that then you will override it to IPC_SOF.

I think the logic should be something like:
```c
if (sof_pci_ipc_type == SOF_IPC_COUNT) {
	sof_pdata->ipc_type = desc->ipc_default;
}else {
	if (sof_pci_ipc_type < 0 || sof_pci_ipc_type >= SOF_IPC_COUNT) {
		dev_err(dev, "invalid request value %d\n", sof_pci_ipc_type);
		return -EINVAL;
	}
	if (!(BIT(sof_pci_ipc_type) & desc->ipc_supported_mask)) {
		dev_err(dev, "invalid request value %d, supported mask is %#x\n",
			sof_pci_ipc_type, desc->ipc_supported_mask);
		return -EINVAL;
	}
	sof_pdata->ipc_type = sof_pci_ipc_type;
	dev_info(dev, "overriding default IPC %d to requested %d\n", desc->ipc_default, sof_pci_ipc_type);
}
```

Reply:
Now I get what you were saying @ujfalusi 

I took your suggestion but used -1 as default for consistency with other kernel module parameters.